mvpoly/src/prime.rs

```diff
@@ -449,8 +449,9 @@ impl<F: PrimeField, const N: usize, const D: usize> MVPoly<F, N, D> for Dense<F,
         unimplemented!()
     }
 
-    fn modify_monomial_with_scalar(&mut self, scalar: F) {
-        self[0] = scalar;
+    fn modify_monomial(&mut self, _exponents: [usize; N], coeff: F) {
+        // Directly modify the coefficient at index 0
+        self[0] = coeff;
```

Comment:
This is not correct. It does require a bit of computation to set correctly the coefficients (computation on the powers).